var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -194,6 +196,10 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
     conflicts("+cuda", when="std=17 ^cuda@:10")
     conflicts("+cuda", when="std=20")
 
+    # SYCL and OpenMPTarget require C++17
+    conflicts("+sycl", when="std=14")
+    conflicts("+openmptarget", when="std=14")
```

Comment:
Given the comment, perhaps this should be a for loop over ["11", "14"] to add conflicts for both variants for both "std=11" and "std=14".

Reply:
Yes, I was asking about this in the slack channel in case there is an elegant way to do this, here is the thread: https://spackpm.slack.com/archives/C5W7NKZJT/p1652887450464869

Reply:
Since none of these backends are available before the 3.3 release that enforced C++14, I didn't add more checks here.
I can do if you really want me to, though (although I don't know the most elegant syntax).

Reply:
FWIW.  Given the number of `devices_variants` and way they are set, I recommend adding a `for` loop as I suggested above or Asher's approach (https://spackpm.slack.com/archives/C5W7NKZJT/p1652890519156749?thread_ts=1652887450.464869&cid=C5W7NKZJT).

In other words, likely simpler to add the conflicts than modify the structure.